lib/features/children/edit_child/cubit/edit_child_cubit.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app/features/children/create_child/mixins/child_giving_allowance_validator.dart';
+import 'package:givt_app/features/children/create_child/mixins/child_name_validator.dart';
+import 'package:givt_app/features/children/create_child/repositories/create_child_repository.dart';
+import 'package:givt_app/features/children/details/models/profile_ext.dart';
+import 'package:givt_app/features/children/edit_child/models/edit_child.dart';
+import 'package:givt_app/l10n/l10n.dart';
+
+part 'edit_child_state.dart';
+
+class EditChildCubit extends Cubit<EditChildState>
+    with ChildNameValidator, ChildGivingAllowanceValidator {
+  EditChildCubit(
+    this._createChildRepository,
+    this._locals,
+    this._profileDetails,
+  ) : super(const EditChildInitialState()) {
+    Future.delayed(Duration.zero, () {
+      emit(
+        EditChildInputState(
+          profileDetails: _profileDetails,
+          child: EditChild.fromProfileDetails(_profileDetails),
+        ),
+      );
+    });
+  }
+
+  final CreateChildRepository _createChildRepository;
+  final AppLocalizations _locals;
```

Comment:
Oh, not sure if this is something we should do. Why do we need it here and not do it based on the state in the build method?

Reply:
I use it for getting errors from validators